### PR TITLE
Move NavigatorCardBody to a new file

### PR DESCRIPTION
**Pull request by jelly**
This makes `app.jsx` two times smaller then it was and we can now easier port the navigatorCardBody to Typescript.

### CODE REVIEW

**Review by allisonkarlitskaya — COMMENTED** [review]
Can we call the filename like `navigator-card-body` like the Cockpit style?

**Review by allisonkarlitskaya — APPROVED** [review]
happy bots = happy lis

File: src/navigator-card-body.jsx [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+            ? (a.name.toLowerCase() < b.name.toLowerCase()
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    case "za":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.name.toLowerCase() > b.name.toLowerCase()
+                ? -1
+                : 1)
+            : compareFileType(a, b);
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:60]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+    case "last_modified":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.mtime > b.mtime
+                ? -1
+                : 1)
+            : compareFileType(a, b);
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:66]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+    case "first_modified":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.mtime < b.mtime
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    default:
+        break;
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:74]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+            const total = boxes.length;
+            const firstOffset = boxes[0].offsetTop;
+            while (++i < total && boxes[i].offsetTop === firstOffset);
+            setBoxPerRow(i);
+        }
+    }
+
+    const onDoubleClickNavigate = useCallback((file) => {
+        const newPath = [...path, file.name].join("/");
+        if (file.type === "dir" || file.to === "dir") {
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:118]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+
+    useEffect(() => {
+        const onKeyboardNav = (e) => {
+            if (e.key === "ArrowRight") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = selectedIdx < sortedFiles.length - 1
+                        ? selectedIdx + 1
+                        : 0;
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:147]: These 7 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+
+                    return [sortedFiles[newIdx]];
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:149]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+                });
+            } else if (e.key === "ArrowLeft") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = selectedIdx > 0
+                        ? selectedIdx - 1
+                        : sortedFiles.length - 1;
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:157]: These 7 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+
+                    return [sortedFiles[newIdx]];
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:159]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+                });
+            } else if (e.key === "ArrowUp") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = Math.max(selectedIdx - boxPerRow, 0);
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:165]: These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 10]
```diff
@@ -0,0 +1,302 @@
...
+
+                    return [sortedFiles[newIdx]];
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:167]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 11]
```diff
@@ -0,0 +1,302 @@
...
+                });
+            } else if (e.key === "ArrowDown") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = Math.min(selectedIdx + boxPerRow, sortedFiles.length - 1);
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:173]: These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 12]
```diff
@@ -0,0 +1,302 @@
...
+
+                    return [sortedFiles[newIdx]];
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:175]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 13]
```diff
@@ -0,0 +1,302 @@
...
+                });
+            } else if (e.key === "Enter" && selected.length === 1) {
+                onDoubleClickNavigate(selected[0]);
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:178]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 14]
```diff
@@ -0,0 +1,302 @@
...
+
+    const resetSelected = e => {
+        if (e.target.id === "folder-view" || e.target.id === "navigator-card-body") {
+            setSelected([]);
+        }
+    };
+
+    const handleClick = (ev, file) => {
+        if (ev.detail > 1) {
+            onDoubleClickNavigate(file);
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:209]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**